backend/wayland/output.c

```diff
@@ -186,6 +186,11 @@ static void wlr_wl_output_destroy(struct wlr_output *wlr_output) {
 		wl_callback_destroy(output->frame_callback);
 	}
 
+	struct wlr_wl_pointer *pointer = output->backend->wlr_wl_pointer;
+	if (pointer && pointer->current_output == output) {
+		pointer->current_output = NULL;
+	}
```

Comment:
Yeah I agree that this isn't great but I don't have any better ideas

Reply:
The pattern we normally use here is for the `wlr_wl_pointer` to have a listener for the output destroy signal and run its own cleanup code when that event is emitted.

Reply:
That is what I first wanted to do, but there can be multiple outputs - how can you make a listener for an arbitrary number of outputs? Have a listener on output add and allocate destroy ones whenever an output is added seems overkill?

Reply:
You only need a listener for whatever output `current_output` points to.

Reply:
So change that listener everytime current output changes? I'm not sure I like evolving listeners much, but if you insist I will...

Reply:
Just make sure there is a setter and do all the listener logic in the setter. That is how we do it in many other places in the project.

Reply:
It's possible, but I don't see anywhere else where that would be done. Could you point me at a place? (I don't need an example, I want a precedent because that's the argument you push forward. Give another argument like "do it because I said so" and I will)
I did this search: `git grep -C 4 wl_signal_add | grep -C 5 wl_signal_remove
`

Reply:
Another example is `display_destroy`.

Reply:
Done in a force-push. No longer need to remove wl_pointer as well, since I don't need to add wlr_wl_pointer to the backend. Tested and it seems to work, I dropped the level of "pointer motion event without current output" because we seem to be missing an enter event sometimes (rootston-in-rootston, so probably a bug in the parent rootston that I couldn't reproduce reliably, set `WLR_WL_OUTPUTS` to something >1 and close outputs to test)
<bike-shedding>I went over the 107 uses of `display_destroy` without seeing anywhere that seems to do that, but meh, I don't care anymore.</bike-shedding>